File: lab3/blink.py
# CS326 Lab 3
# Blinking LED

from gpiozero import LED
import time

# Create LED object using GPIO pin 16
led = LED(16)
DELAY = 0.5

# Blink the LED 20 times
for count in range(20):
    led.on()
    print('LED: on')
    time.sleep(DELAY)
    led.off()
    print('LED: off')
    time.sleep(DELAY)

led.close()
print("Done!")

File: lab3/blink.c
/* CS326 Lab 3
   Blinking LED */
#include <stdio.h>
#include <pigpio.h>
#include <unistd.h>

#define LED 16
#define DELAY 1

int main (int argc, char *argv[])
{
   if (gpioInitialise()<0) return 1;

   gpioSetMode(LED, PI_OUTPUT);
   while (1)
   {
      gpioWrite(LED, PI_ON);
      printf("LED ON\n");
      sleep(DELAY);
      gpioWrite(LED, PI_OFF);
      printf("LED OFF\n");
      sleep(DELAY);
   }
   gpioTerminate();
   return 0;
}
